muesli/models.py

```diff
@@ -371,7 +371,11 @@ def getResultsForStudent(self, student):
         for points in pointsQuery.all():
             results[points.exercise_id] = {'points': points.points,
                                            'exercise': points.exercise}
-        results['sum'] = sum([x for x in [r['points'] for r in list(results.values())] if x])
+        nonNullPoints = [x for x in [r['points'] for r in list(results.values())] if x]
+        if len(nonNullPoints) > 0:
+            results['sum'] = sum(nonNullPoints)
+        else:
+            results['sum'] = None
```

Comment:
I am not sure whether this separate handling of the cases makes sense, for two reasons:

1. Does a default of `None` really make sense here? IMO `0` would make more sense
2. As per 1), `sum([])` (Sum of the array with `len(array) == 0`) yields `0`, so we might aswell directly use it! 😄

Reply:
This is in fact intended as the code in `viewsLecture.py` distiguishes between ungraded and graded sheets with zero points by checking whether `result['sum']` is `None` :)